### ISSUE TITLE
Minecraft server continually crashing

### ISSUE THREAD

**Issue by claine-d** [itzg/docker-minecraft-bedrock-server#122]
Hi - not sure if this is an issue caused by running in Docker or with the Minecraft Server software itself.  

The server crashes constantly now after about 30s to a minute.  I get a crashreporter key and then what looks like C code shortdumping.

`Session ID: 5f575187-1a9e-4ad1-8bfb-21986fcefe29
mc     | Commit hash:
mc     | Build id: development
mc     | CrashReporter Key: 8c4937c1-64cb-3532-a8dc-1deb28f67293
mc     |
mc     | Crash
mc     | [INFO]         at std::__shared_count<[__gnu_cxx::_Lock_policy]2>::__shared_count[std::__weak_count<[__gnu_cxx::_Lock_policy]2> const&, std::nothrow_t] (UnknownFile:?)
mc     |        at std::__shared_ptr<POIInstance, [__gnu_cxx::_Lock_policy]2>::__shared_ptr[std::__weak_ptr<POIInstance,[__gnu_cxx::_Lock_policy]2> const&, std::nothrow_t] (UnknownFile:?)
mc     |        at std::shared_ptr<POIInstance>::shared_ptr[std::weak_ptr<POIInstance> const&, std::nothrow_t] (UnknownFile:?)
mc     |        at std::weak_ptr<POIInstance>::lock[] const (UnknownFile:?)
mc     |        at Village::getBedPOICount[] const (UnknownFile:?)
mc     |        at Village::tick[Tick, BlockSource&] (UnknownFile:?)
mc     |        at VillageManager::tickVillages[Tick const&, Vec3 const&, BlockSource&] (UnknownFile:?)
mc     |        at ServerPlayer::tickWorld[Tick const&] (UnknownFile:?)
mc     |        at std::function<bool [Player&]>::operator[][Player&] const (UnknownFile:?)
mc     |        at Level::forEachPlayer[std::function<bool [Player&]>] (UnknownFile:?)
mc     |        at Level::tick[] (UnknownFile:?)
mc     |        at ServerLevel::tick[] (UnknownFile:?)
mc     |        at GameSession::tick[] (UnknownFile:?)
mc     |        at Minecraft::tickSimtime[int, int] (UnknownFile:?)
mc     |        at Minecraft::update[] (UnknownFile:?)
mc     |        at ServerInstance::_update[] (UnknownFile:?)
mc     |        at clone (UnknownFile:?)
mc     | 10b70601-d8af-4e96-ab99-cd4b7cc3e5c3mc exited with code 1
`

Thanks for looking

**Comment by itzg** [comment]
Sorry for the delay replying to this. It seems like something that would need to be reported to and fixed by Mojang via their bug tracker.

**Comment by claine-d** [comment]
I did open up a bug with minecraft directly.  I think a chunk was corrupted and now if you walk into that chunk the server crashes.  I don't think there's anything else from the docker container side of it.  Thanks

**Comment by ChristopherHaws** [comment]
@itzg Someone posted a workaround on the Minecraft Bug tracker for this issue (they are specifically using your docker image) and their fix seems to be working for them. Is there anyway we can get that patch applied to the docker file?

https://bugs.mojang.com/browse/BDS-10666?focusedCommentId=893349&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-893349

Thanks for the awesome docker image! :)

**Comment by itzg** [comment]
So it looks like [in the final comment](https://bugs.mojang.com/browse/BDS-10666?focusedCommentId=893453&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-893453), that an apt upgrade of libc might do it. I could add an "apt upgrade" of everything, just to be thorough.

**Comment by itzg** [comment]
...well, libc6 is already at the latest since this is all that gets listed:

```
root@6b7389b6b8d3:/data# apt list --upgradable
Listing... Done
libp11-kit0/stable 0.23.15-2+deb10u1 amd64 [upgradable from: 0.23.15-2]
tzdata/stable-updates 2020e-0+deb10u1 all [upgradable from: 2020d-0+deb10u1]
```

The base image may have upgraded its packages since the original issue.

**Comment by ChristopherHaws** [comment]
Awesome, I will give it a go tonight and let you know if the new image fixes this issue. Thanks for the quick turnaround time :)

**Comment by ChristopherHaws** [comment]
Unfortunately this issue still happens. I had to switch over to using the windows version of the server software which seems to be way less buggy sadly. :(

Thanks for your help!

**Comment by ChristopherHaws** [comment]
Just an FYI, it looks like this issue was fixed in the latest bedrock edition server software that was released 3 days ago (1.16.201.03):
https://bugs.mojang.com/browse/BDS-10666?focusedCommentId=917681&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-917681
